### ISSUE TITLE
Make default simulator a newer iPhone than the old iPhone SE

### ISSUE THREAD

**Issue by roryabraham** [Expensify/App#26224]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Run `npm run ios`

## Expected Result:
The iOS app should open in a newer simulator, say an iPhone 14 at this moment in time.

## Actual Result:
By default XCode opens an older iPhone SE simulator, which isn't really representative of are most common target audiences.

## Workaround:
- Find the deviceID of the simulator you want in XCode then run `npm run ios -- --device="deviceID"`
- Open XCode, select the target device and run the app with the play button in the UI.

## Platforms:
Which of our officially supported platforms is this issue occurring on?
- [x] iOS / native

**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1692316899451739

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by b4s36t4** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Make default simulator a newer iPhone than the old iPhone SE
### What is the root cause of that problem?
No RC
### What changes do you think we should make in order to solve the problem?
Instead of adding a default platform why not let the developer choose the device they want to work.

Sometimes some devices give different layout depends on dimensions of device.

To solve this we need to add a parameter `--list-devices` which shows the list of available iOS devices that user can select and open with.

Parameter need to be added here https://github.com/Expensify/App/blob/4675db5348e305daca320f91066476344191d8d2/package.json#L14.


### What alternative solutions did you explore? (Optional)
NA

<img width="767" alt="Screenshot 2023-08-29 at 9 27 49 PM" src="https://github.com/Expensify/App/assets/59088937/cddff950-d77c-4169-bb8e-879e9c734979">

> This would show all the available devices.

**Comment by roryabraham** [comment]
Sorry @GItGudRatio, but I think that @b4s36t4's proposal is a superior DX because it offers greater flexibility and also makes it easy to use whatever sim you want.

**Comment by roryabraham** [comment]
Yeah, I think if people really want to avoid the extra click they can always provide a `--device` argument manually or via a custom terminal alias. imo the problem we're solving here is that the default is an old / uncommon iPhone.

I feel good about the solution we've selected here 👍🏼

**Comment by greg-schroeder** [comment]
Sorry I've been OOO with an illness, will process this today

**Comment by greg-schroeder** [comment]
Paid.

**Comment by b4s36t4** [comment]
Any urgency bonus applies here? I think this only got delayed because of merge freeze @roryabraham ?
